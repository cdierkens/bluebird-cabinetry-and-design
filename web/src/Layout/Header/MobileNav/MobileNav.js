import { Link } from "gatsby";
import { PropTypes } from "prop-types";
import React, { useState } from "react";
import logo from "../../../logo.png";
import styles from "./MobileNav.module.css";

const MobileNav = ({ children }) => {
  const [isOpen, setIsOpen] = useState();

  return (
    <div className={styles.MobileNav}>
      <nav className="grid grid-cols-3 sm:grid-cols-6">
        <div className="col-span-2 sm:col-span-5">
          <Link className="m-3 block" to="/">
            <img className={styles.Logo} src={logo} alt="Bluebird Logo" />
          </Link>
        </div>

        <div className="col-start-3 sm:col-start-6 items-center justify-end flex">
          <button
            onClick={() => setIsOpen((open) => !open)}
            className="flex items-center px-3 py-2 border rounded text-blue-dark hover:bg-white m-3"
          >
            <svg
              className="fill-current h-3 w-3"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>Menu</title>
              <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" />
            </svg>
          </button>
        </div>
        {isOpen && (
          <div className="col-span-3 sm:col-span-6 flex flex-col">
            {children}
          </div>
        )}
      </nav>
    </div>
  );
};

MobileNav.propTypes = {
  children: PropTypes.node,
};

export default MobileNav;
