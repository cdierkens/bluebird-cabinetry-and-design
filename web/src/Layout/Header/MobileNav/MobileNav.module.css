.MobileNav {
  /* Plopped styles for MobileNav */
}

.Logo {
  @apply bg-blue-dark rounded-md px-1 h-24;
}
