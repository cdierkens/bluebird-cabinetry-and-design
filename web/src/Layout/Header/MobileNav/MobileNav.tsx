import { Link } from "gatsby";
import React, { useState } from "react";
import logo from "../../../images/logo.png";
import * as styles from "./MobileNav.module.css";

const MobileNav: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
  const [isOpen, setIsOpen] = useState<boolean>(false);

  return (
    <div className={styles.MobileNav}>
      <nav className="grid grid-cols-2 sm:grid-cols-6">
        {!isOpen && (
          <div className="col-span-2 sm:col-span-5 absolute">
            <Link className="m-3 block" to="/">
              <img className={styles.Logo} src={logo} alt="Bluebird Logo" />
            </Link>
          </div>
        )}

        <div className="col-start-3 sm:col-start-6 items-center justify-end flex">
          <button
            onClick={() => setIsOpen((open) => !open)}
            className="flex items-center px-3 py-2 border rounded text-blue-dark hover:bg-white m-3"
          >
            <svg
              className="fill-current h-3 w-3"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>Menu</title>
              <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" />
            </svg>
          </button>
        </div>
        {isOpen && (
          <div className="col-span-3 sm:col-span-6 flex flex-col">
            {children}
          </div>
        )}
      </nav>
    </div>
  );
};

export default MobileNav;
