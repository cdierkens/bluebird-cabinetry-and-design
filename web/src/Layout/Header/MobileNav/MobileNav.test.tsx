import { render } from "@testing-library/react";
import React from "react";
import MobileNav from "./MobileNav";

it("matches snapshot", () => {
  const { asFragment } = render(<MobileNav />);

  expect(asFragment()).toMatchSnapshot();
});
