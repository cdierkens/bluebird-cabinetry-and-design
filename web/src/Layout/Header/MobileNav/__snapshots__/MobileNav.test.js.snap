// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MobileNav"
  >
    <nav
      class="grid grid-cols-3 sm:grid-cols-6"
    >
      <div
        class="col-span-2 sm:col-span-5"
      >
        <a
          class="m-3 block"
          href="/"
        >
          <img
            alt="Bluebird Logo"
            class="Logo"
            src="test-file-stub"
          />
        </a>
      </div>
      <div
        class="col-start-3 sm:col-start-6 items-center justify-end flex"
      >
        <button
          class="flex items-center px-3 py-2 border rounded text-blue-dark hover:bg-white m-3"
        >
          <svg
            class="fill-current h-3 w-3"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Menu
            </title>
            <path
              d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
            />
          </svg>
        </button>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
